docs/articles/vault.md

```diff
@@ -1,43 +1,194 @@
+the CSI provider.
 
-For integrations utilizing the [sidecar
-injector](https://developer.hashicorp.com/vault/docs/platform/k8s/injector) you
-can try to set the appropriate annotations by adapting the example
-configurations below for your needs. If you encounter issues please reach out to
-our enterprise support.
+:::warning
 
-## Configurations for sidecar injector
+Prebuilt and container executors are deprecated so we do not officially support
+their usage with Vault, please migrate to workflows.
 
-With workflows, you can configure pod annotations both per workflow and
-globally:
+:::
 
-Chart `testkube`:
+## Sidecar injector
+
+Vault's sidecar injector injects an init container to fetch secrets on
+initialization and a sidecar container to periodically update that secret.
+Either container can be disabled, but we will focus on the configuration which
+enables both.
+
+
+To make the workflow workloads compatible with Vault's injector, we can utilize
+this workflow template:
+
+```yaml
+apiVersion: testworkflows.testkube.io/v1
+kind: TestWorkflowTemplate
+metadata:
+  name: vault-secret-injection
+spec:
+  config:
+    port:
+      type: string
+      default: "8200"
+  pod:
+    annotations:
+      vault.hashicorp.com/agent-inject: "true"
+      vault.hashicorp.com/agent-init-first: "true"
+      vault.hashicorp.com/agent-enable-quit: "true"
+      vault.hashicorp.com/agent-cache-listener-port: "{{ config.port }}"
+  after:
+  - name: 'Send quit signal to Vault agent'
+    container:
+      image: busybox
+    condition: always
+    shell: 'while ! wget --post-data "" -O - http://localhost:{{ config.port }}/agent/v1/quit; do sleep 1; done'
+```
+
+This workflow template:
+
+- Enables the sidecar injection.
+- Makes sure that the Vault init container is ahead of other init containers. This is
+  required to have access to secrets within workflow steps as workflows run in
+sequence within init containers.
+- Configures the agent server running within the sidecar.
+- Calls the agent's quit endpoint after all steps of the workflow have
+  completed. Otherwise, the sidecar container would never exit keeping the
+workflow running indefinitely.
+
+One can then reuse this workflow template in workflows requiring secret
+injection (example builds on Vault's great tutorial
+[here](https://developer.hashicorp.com/vault/tutorials/kubernetes/kubernetes-sidecar)):
+
+```yaml
+apiVersion: testworkflows.testkube.io/v1
+kind: TestWorkflow
+metadata:
+  name: check-vault-secret-injection
+spec:
+  pod:
+    # highlight-next-line
+    serviceAccountName: test-vault
+    labels:
+      app: workflow-vault
+    annotations:
+      # highlight-start
+      vault.hashicorp.com/agent-inject-secret-database-config.txt: internal/data/database/config
+      vault.hashicorp.com/role: internal-app
+      # highlight-end
+  # highlight-start
+  use:
+    - name: 'vault-secret-injection'
+  # highlight-end
+  container:
+    image: alpine:latest
```

Comment:
Is that required?

Reply:
I guess I can specify this at the step level to be more clear here. Does that make more sense?

Reply:
I'm just thinking if it's needed at all, the shell script should work on the basic image as well, unless `alpine` has something specific

Reply:
Oh, ok - I did not try the default. Let me try.

Reply:
It works, switched to default image.